0000-hashed-timelock-agreements.md

```diff
@@ -0,0 +1,120 @@
+
+Before reading this document, it is important to understand how Interledger isolates participants from risks posed by indirect connections. The tradeoffs listed here are only the concern of parties that are _directly connected_. For more on the security model, see the [Interledger Architecture](./0001-interledger-architecture/0001-interledger-architecture.md).
+
+## Table of Contents
+
+1. [Background on Hashed-Timelock Contracts (HTLCs)](#background-on-hashed-timelock-contracts-htlcs)
+2. [Hashed-Timelock Agreements (HTLAs)](#hashed-timelock-agreements)
+3. [HTLAs Without Ledger Support](#htlas-without-ledger-supoprt)
+4. [Spectrum of HTLA Types](#spectrum-of-htla-types)
+   1. [Conditional Payment Channels (with HTLCs)](#conditional-payment-channels-with-htlcs)
+   2. [On-Ledger Holds/Escrow (using HTLCs)](#on-ledger-holds-escrow-using-htlcs)
+   3. [Simple Payment Channels](#simple-payment-channels)
+   4. [Trustlines](#trustlines)
+5. [Conclusion: Integrating All the Ledgers](#conclusion-integrating-all-the-ledgers)
+
+_____
+
+## Background on Hashed-Timelock Contracts (HTLCs)
+
+A [Hashed-Timelock Contract (HTLC)](https://en.bitcoin.it/wiki/Hashed_Timelock_Contracts) is a conditional transfer where the condition is enforced by the ledger. It is a concept from the Bitcoin community that is used in the [Lightning Network](https://lightning.network).
+
+When the transfer is "prepared", the sender's funds are put on hold by the ledger, pending the fulfillment of a predefined condition. The condition is a hashlock, or the digest of a cryptographic hash function, such as SHA-256 in Lightning and Interledger. The "contract" stipulates that the recipient may claim the funds by presenting a valid preimage of the hash digest before a given timeout. After the timeout, the funds are automatically returned to the sender. This is the Hashed-Timelock Contract.
+
+HTLCs are enforced by the ledger, so the parties transacting only need to trust the ledger to correctly execute the contract. However, this mechanism requires support from the ledger and only works with ledgers that implement hashlocks and timeouts.
+
+Interledger is designed to work with all ledgers, so it must support both ledgers with and without support for hashlocks and timeouts.
+
+## Hashed-Timelock Agreements (HTLAs)
+
+A contract is a certain type of agreement that is enforced by a third party. Hashed-Timelock Agreements (HTLAs) generalize the idea of HTLCs to include both agreements that are and are not enforced by a ledger. The concept was originally proposed in [this thread on the Interledger mailing list](https://lists.w3.org/Archives/Public/public-interledger/2017Jun/0009.html).
```

Comment:
> generalize the idea of HTLCs to include both agreements that are and are not enforced by a ledger.

I think it's clearer to say "to include agreements that are not enforced by a ledger"